src/middlewares.rs

```diff
@@ -52,7 +52,7 @@ impl ApiErrorResponse {
     pub fn from_fail(fail: &dyn Fail) -> ApiErrorResponse {
         let mut messages = vec![];
 
-        for cause in Fail::iter_chain(fail) {
+        for cause in <dyn Fail>::iter_chain(fail) {
```

Comment:
btw, this is:
https://github.com/rust-lang/rust/pull/82868 (fixing https://github.com/rust-lang/rust/issues/65371)

It seemed weird at first, but does make sense. Also looks like that these will become hard errors in Rust 2021 soon: https://github.com/rust-lang/rust/pull/83213